src/fpd/enrichment.js

```diff
@@ -78,6 +95,34 @@ function removeUndef(obj) {
   return getDefinedParams(obj, Object.keys(obj))
 }
 
+export async function tryToGetCdepLabel(cb = getCookieDeprecationLabel) {
+  let cdep;
+  const consentData = gdprDataHandler.getConsentData();
```

Comment:
This should do a check on the `accessDevice` activity instead (somewhat like [this](https://github.com/prebid/Prebid.js/blob/30ac7f5dff10cd5ce9841c4451641d797e888d4f/src/storageManager.js#L32-L34)). 

Normally that would be enough - the TCF logic (which is just one of the ways storage is gated) is attached to `accessDevice` [in its own module](https://github.com/prebid/Prebid.js/blob/30ac7f5dff10cd5ce9841c4451641d797e888d4f/modules/gdprEnforcement.js#L228-L234).

However for this case we decided that the TCF logic should be changed to check for publisher consent (`publisher.consents[1]` in the parsed TCF data), apparently that's what we should have been doing from the beginning, but we need to wait for 9 to avoid breaking people. Paging also @patmmccann  since I can't find a summary of what was decided, I think we may also want to add `transmitUfpd` (purpose 4) checks.

For this PR that means:

 - the TCF logic linked above should be updated to look for `publisher.*`. I would put in a special case for a new component name, e.g. if in here you check `isActivityAllowed(ACTIVITY_ACCESS_DEVICE, activityParams(MODULE_TYPE_PREBID, 'cdep'))`, then [here](https://github.com/prebid/Prebid.js/blob/30ac7f5dff10cd5ce9841c4451641d797e888d4f/modules/gdprEnforcement.js#L231) you can have a special case for when `params[ACTIVITY_PARAM_COMPONENT_NAME] === 'cdep'`. That would then get conflated into the existing special case for `MODULE_TYPE_PREBID` in prebid 9.
 - If we need purpose 4 checks it should be enough to add `device.ext.cdep` to [this list](https://github.com/prebid/Prebid.js/blob/30ac7f5dff10cd5ce9841c4451641d797e888d4f/src/activities/redactor.js#L11-L21)

Reply:
Update:

- The new TCF logic should (in version 9) replace [this](https://github.com/prebid/Prebid.js/blob/30ac7f5dff10cd5ce9841c4451641d797e888d4f/modules/gdprEnforcement.js#L186-L194) so that if gvlid is `VENDORLESS_GVLID`, instead of looking at [`vendorData.purpose`](https://github.com/prebid/Prebid.js/blob/30ac7f5dff10cd5ce9841c4451641d797e888d4f/modules/gdprEnforcement.js#L170-L171)  and switching off the check on `vendorData.vendor`, we check `vendorData.publisher` instead. So my suggestion is to introduce a second, temporary gvlid placeholder (maybe `FIRST_PARTY_GVLID`); use it to implement the new `publisher` logic, and update [this](https://github.com/prebid/Prebid.js/blob/30ac7f5dff10cd5ce9841c4451641d797e888d4f/modules/gdprEnforcement.js#L113-L115) to return it only for the cookie deprecation label for now - then in 9 we can get rid of the existing `VENDORLESS_GVLID` logic.
- We do want `transmitUfpd` / purpose 4 checks, so `device.ext.cdep` should be added to the [ORTB fields that are redacted out when that is denied](https://github.com/prebid/Prebid.js/blob/30ac7f5dff10cd5ce9841c4451641d797e888d4f/src/activities/redactor.js#L11-L21)